test/Libraries/DynamoMSOfficeTests/OpenXmlTests.cs

```diff
@@ -0,0 +1,770 @@
+
+            var stringInput = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<CoreNodeModels.Input.StringInput>();
+            stringInput.Value = filePath;
+
+            var readNode = ViewModel.Model.CurrentWorkspace.Nodes.First(n => n.Name == "Data.ImportOpenXml");
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(readNode.CachedValue.IsCollection);
+            var data = new object[]
+            {
+                new object[] {999, 999, 999},
+                new object[] {999, 999, 999},
+                new object[] {999, 999, 999}
+            };
+            AssertPreviewValue(readNode.GUID.ToString(), data);
+        }
+
+        [Test]
+        public void CanOverWritePartiallyToExcel()
+        {
+            // Copy the file so we don't write over the original
+            string excelFilePath = Path.Combine(TestDirectory, @"core\excel\overwriteFilePartial.xlsx");
+            var filePath = Path.Combine(TempFolder, $"{nameof(CanOverWritePartiallyToExcel)}_output.xlsx");
+            File.Copy(excelFilePath, filePath);
+
+            string openPath = Path.Combine(TestDirectory, @"core\excel\OverWritePartialExcelSheet_OpenXml.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var stringInput = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<CoreNodeModels.Input.StringInput>();
+            stringInput.Value = filePath;
+
+            var readNode = ViewModel.Model.CurrentWorkspace.Nodes.First(n => n.Name == "Data.ImportOpenXml");
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(readNode.CachedValue.IsCollection);
+            var data = new object[]
+            {
+                new object[] {1, 1, 1, 1, 1},
+                new object[] {1, 999, 999, 999, 1},
+                new object[] {1, 999, 999, 999, 1},
+                new object[] {1, 999, 999, 999, 1},
+                new object[] {1, 1, 1, 1, 1}
+            };
+            AssertPreviewValue(readNode.GUID.ToString(), data);
+        }
+
+        [Test]
+        public void CanWriteNullValuesToExcel()
+        {
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNullValuesToExcel_OpenXml.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filePath = Path.Combine(TempFolder, $"{nameof(CanWriteNullValuesToExcel)}_output.xlsx");
+            var stringNode = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<CoreNodeModels.Input.StringInput>();
+            stringNode.Value = filePath;
+
+            var readNode = ViewModel.Model.CurrentWorkspace.Nodes.First(n => n.Name == "Data.ImportOpenXml");
+            ViewModel.HomeSpace.Run();
+
+            Assert.IsTrue(readNode.CachedValue.IsCollection);
+            var list = readNode.CachedValue.GetElements();
+            Assert.AreEqual(0, list.Count());
+        }
+
+        [Test]
+        public void CanWriteNullValuesToExcel1()
```

Comment:
Can you comment on this test the diff with the test above or use longer name to make it self explain? From reading the code, we are making sure not single value but `null` in the middle of array can be written?

Reply:
I think I can, let me take a look at it to make sure. The "undocumented" aspect is inherited from the existing tests which I copied, but let me see if I can improve that here.

Reply:
I renamed the tests a little bit to remark the difference. The previous test writes `null` as data, while this one includes `null` as a value in a list.

Reply:
@mmisol Great! I will leave some review period today for team and come back tomorrow to check more comments, if not expect our team to merge and bang on these this sprint!